plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/ui/ide/ExtendedFileEditorsPreferencePage.java

```diff
@@ -0,0 +1,152 @@
+ * 
+ * 
+ * @author rhopp
+ *
+ */
+
+public class ExtendedFileEditorsPreferencePage extends PreferencePage {
+
+    public ExtendedFileEditorsPreferencePage() {
+	super("General", "Editors", "File Associations");
+    }
+
+    /**
+     * Returns all file types.
+     * 
+     * @return all file types.
+     */
+
+    public List<String> getFileTypes() {
+	List<String> resultList = new ArrayList<String>();
+	DefaultTable fileTypesTable = new DefaultTable();
+	List<TableItem> fileTypeItems = fileTypesTable.getItems();
+	for (TableItem tableItem : fileTypeItems) {
+	    resultList.add(tableItem.getText());
+	}
+	return resultList;
+    }
+
+    /**
+     * Selects desired file type.
+     * 
+     * @param fileType
+     */
+
+    public void selectFileType(String fileType) {
+	DefaultTable fileTypesTable = new DefaultTable();
+	TableItem item = fileTypesTable.getItem(fileType);
+	item.select();
+    }
+
+    /**
+     * Adds new file type.
+     * 
+     * @param fileType
+     */
+    public void addFileType(String fileType) {
+	new PushButton("Add...").click();
+	new DefaultShell("Add File Type");
+	new LabeledText("File type:").setText(fileType);
+	new OkButton().click();
+    }
+
+    /**
+     * Removes given file type.
+     * 
+     * @param fileType
+     */
+    public void removeFileType(String fileType) {
+	selectFileType(fileType);
+	new PushButton("Remove").click();
+    }
+
+    /**
+     * Returns list of editors for currently selected file type.
+     * 
+     * @return
+     */
+    public List<String> getAssociatedEditors() {
+	List<String> resultList = new ArrayList<String>();
+	DefaultTable editorsTable = new DefaultTable(1);
+	List<TableItem> editors = editorsTable.getItems();
+	for (TableItem editorTableItem : editors) {
+	    resultList.add(editorTableItem.getText());
+	}
+	return resultList;
+    }
+
+    /**
+     * Returns list of editors for given file type.
+     * 
+     * @param fileType
+     * @return
+     */
+    public List<String> getAssociatedEditorForFileType(String fileType) {
+	selectFileType(fileType);
+	return getAssociatedEditors();
+    }
+
+    /**
+     * Selects given editor.
+     * 
+     * @param editor
+     */
+    public void selectAssociatedEditor(String editor) {
+	DefaultTable editorsTable = new DefaultTable(1);
+	TableItem item = editorsTable.getItem(editor);
+	item.select();
+    }
+
+    /**
+     * Makes given editor default for currently selected file type.
+     * 
+     * @param editor
+     */
+    public void makeEditorDefault(String editor) {
+	selectAssociatedEditor(editor);
+	new PushButton("Default").click();
+    }
+
+    /**
+     * Adds new internal editor for currently selected file type.
+     * 
+     * @param editor
+     */
+    public void addInternalEditor(String editor) {
+	new PushButton(1, new WithMnemonicTextMatcher("Add...")).click();
+	new DefaultShell("Editor Selection");
+	DefaultTree tree = new DefaultTree();
+	new DefaultTreeItem(tree, editor).select();
```

Comment:
no need to instantiate tree. You can use only new DefaultTreeItem(editor).select();